client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -22,80 +22,142 @@ const siteTypeOptions: Record< SiteType, string > = {
+		setLastSelectedQuestion( null );
+	};
+
+	const openSiteType = () => {
+		setLastSelectedQuestion( 'siteType' );
+	};
+
+	const selectSiteVertical = () => {
+		setLastSelectedQuestion( 'siteVertical' );
+	};
+
+	const selectSiteTitle = () => {
+		setLastSelectedQuestion( 'siteTitle' );
+	};
+
+	const isSelected = questionType => {
+		if ( lastSelectedQuestion ) {
+			return questionType === lastSelectedQuestion;
+		} else if ( isEmpty( siteType ) ) {
+			return questionType === 'siteType';
+		} else if ( isEmpty( siteVertical ) ) {
+			return questionType === 'siteVertical';
+		} else if ( questionType === 'siteTitle' ) {
+			return true;
+		}
+		return false;
+	};
+
 	return (
-		<div className="onboarding-block__questions">
-			<h2 className="onboarding-block__questions-heading">
-				{ NO__( "Let's set up your website – it takes only a moment." ) }
-			</h2>
-
-			<div className="onboarding-block__question">
-				<span>{ NO__( 'I want to create a website ' ) }</span>
-				{ ! isFilledFormValue( siteType ) ? (
-					<ul className="onboarding-block__multi-question">
-						{ map( siteTypeOptions, ( label, value ) => (
-							<li key={ value }>
-								<label>
-									<input
-										name="onboarding_site_type"
-										onChange={ updateSiteType }
-										type="radio"
-										value={ value }
-									/>
-									<span className="onboarding-block__multi-question-choice">{ label }</span>
-								</label>
-							</li>
-						) ) }
-					</ul>
-				) : (
-					<div className="onboarding-block__multi-question">
-						<button className="onboarding-block__question-answered" onClick={ resetSiteType }>
-							{ siteTypeOptions[ siteType ] }
-						</button>
+		<div className="onboarding-block__acquire-intent">
+			<div className="onboarding-block__questions">
+				<h2 className="onboarding-block__questions-heading">
+					{ isEmpty( siteType ) && NO__( "Let's set up your website – it takes only a moment." ) }
+				</h2>
+
+				<div
+					className={ classNames( {
+						'onboarding-block__question': true,
+						selected: isSelected( 'siteType' ),
+					} ) }
+				>
+					<span>
+						{ isEmpty( siteType )
+							? NO__( 'I want to create a website ' )
+							: NO__( 'Setting up a website ' ) }
+					</span>
+					{ isSelected( 'siteType' ) ? (
+						<ul className="onboarding-block__multi-question">
+							{ map( siteTypeOptions, ( label, value ) => (
+								<li key={ value } className={ value === siteType ? 'selected' : null }>
+									<label>
+										<input
+											name="onboarding_site_type"
+											onChange={ selectSiteType }
+											type="radio"
+											value={ value }
+										/>
+										<span className="onboarding-block__multi-question-choice">{ label }</span>
+									</label>
+								</li>
+							) ) }
+						</ul>
+					) : (
+						<div className="onboarding-block__multi-question">
+							<button className="onboarding-block__question-answered" onClick={ openSiteType }>
```

Comment:
This changes our state handling rather significantly :thinking: It used to be `resetSiteType`, which reset _Redux_ state (which was readable via `isFilledFormValue( siteType )`). Now it only changes component state, which probably means that `isFilledFormValue( siteType )` doesn't work anymore :confused: 

We should try to streamline state handling to make it consistent and intuitive.

Reply:
I think what we vaguely want is:

- Keep Redux (e.g. `resetSiteType`) based handling for emptiness of individual control state
- Add new component state to focus currently active component
- If no component is focused, fall back to 'first' empty component, in order

Reply:
> it used to be resetSiteType, which reset Redux state

yeah, I changed it because... why do we want to reset redux state? the fact that the user open that question again doesn't invalidate their previous choices! That's why ... well, openSiteType doesn't interact with redux at all, because... it's not supposed to do it :) Redux value gets updated when the user changes their selection (using `selectSiteType`, which calls `setSiteType`). But once the user has set up a value, why would we want to reset it?

```
Add new component state to focus currently active component
If no component is focused, fall back to 'first' empty component, in order
```

This is a little bit what the `isSelected` function is doing  (check component state for a user selection, and if not check for the first empty input), but yeah, that logic probably could be written in a more clear way

Reply:
@johnHackworth 
We used that reset together with `isFilledFormValue` to hide all the other sub-steps after the one the user interacted with. This doesn't mean we reset their values too. We reset just the one being currently edited. 
But now, as @ockham is suspecting, the hiding part is no longer working.
When reading this: 
> • Clicking the blue links in previous steps (ex. “with a blog, for a business”) returns to that step, with all input saved for other completed steps.

I understand we should hide the steps below (keeping their selected values in redux) when clicking on the blue link next to _Setting up a website_ text.

Reply:
> This doesn't mean we reset their values too. We reset just the one being currently edited.

oh I know I know, I mean that it doesn't make any sense to reset _that_ value. The user already choosed something there, they may want to click on "next" or just move to another question again without having to re-make their selection. 

> I understand we should hide the steps below (keeping their selected values in redux) when clicking on the blue link next to Setting up a website text.

I don't think this is the case ... I mean, why to hide the other steps, if the user has already seen them? They want to check their previous selection, but then not change it ... they have to be able to move forward to the next step (for example, let's say that a user selects "with a blog", then "Internet Cafe" and then, once they are in the site title step, decide to go back and change it to just "cafe" ... but before doing it, they change opinion again and decide to just go with "internet cafe" ... they have to be able to go back to the site title option without having to re-select anything.

Hiding the 2&3 question makes sense when the user lands in the form, to get a sense of progression ... but once they have seen a question, I don't think we should hide it again, even if they go back

Reply:
> Hiding the 2&3 question makes sense when the user lands in the form, to get a sense of progression ... but once they have seen a question, I don't think we should hide it again, even if they go back

Yeah, consistency with the initial progressing was the motivation here -- to some extent also since it originally allowed to achieve the same UI behavior with less state. (Note that we'd only hide subsequent question if the user hasn't selected anything there yet, i.e. if they're empty; I never wanted to hide them if they've already been answered.)